### PR TITLE
Adds a Heresy phobia, replacing the supernatural phobia on heretic sacrifices

### PR DESCRIPTION
**Pull request by DATA-xPUNGED**
## About The Pull Request

This is a rebase of #72563
This adds a new "Heresy" phobia, which replaces the supernatural phobia on heretic sacrifices. However, sacrificing another heretic does not give them this phobia, but instead removes 3 knowledge points + gives them a different flavour text.
## Why It's Good For The Game

Makes wizards, chaplains, heretics, and cultists able to keep playing the video game as their role when they're sacrificed, while still making the phobia follow it's intended purpose of effectively round removing that person in the heretic's eyes.
## Changelog
:cl:
add: Added a phobia of heresy, which replaces the phobia of the supernatural on heretic sacrifices.
add: Sacrificed heretics won't get this phobia, but instead lose 3 knowledge points.
/:cl:

### CODE REVIEW

**Comment by Profakos** [comment]
Aw dang I never got around to do my version.

**Comment by Helg2** [comment]
Please don't copypaste code from the pr except phobia part, it's just won't work and i'm feeling like a shittone of shame

**Comment by DATA-xPUNGED** [comment]
> Please don't copypaste code from the pr except phobia part, it's just won't work and i'm feeling like a shittone of shame

as it is a rebase, i've pretty much copied everything, i am trying to figure out how to MAKE IT work

**Comment by DATA-xPUNGED** [comment]
> Aw dang I never got around to do my version.

If you have idea on how to help feel free to PR it

File: code/_globalvars/phobias.dm [hunk 1]
```diff
@@ -23,6 +24,7 @@ GLOBAL_LIST_INIT(phobia_types, sort_list(list(
 	"spiders",
 	"strangers",
 	"the supernatural",
+	"heresy",
```
Comment (Helg2) [inline code/_globalvars/phobias.dm:27]: Redundant "heresy"

File: code/_globalvars/phobias.dm [hunk 2]
```diff
@@ -454,6 +488,26 @@ GLOBAL_LIST_INIT(phobia_objs, list(
...
+		/obj/item/codex_cicatrix,
+		/obj/item/reagent_containers/cup/beaker/eldritch,
+		/obj/item/melee/rune_carver,
+		/obj/item/melee/sickly_blade,
+		/obj/item/melee/touch_attack/mansus_fist,
+		/obj/item/toy/eldritch_book,
+		/obj/item/toy/reality_pierce,
+		/obj/structure/destructible/eldritch_crucible,
+		/obj/projectile/curse_hand,
+	)),
```
Comment (Helg2) [inline code/_globalvars/phobias.dm:510]: Redundant and non-alphabetical

File: strings/phobia.json [hunk 3]
```diff
@@ -183,7 +183,28 @@
...
+		"Rust Walker",
+		"Stalker",
+		"Ghoul",
+		"Pierced Reality",
+		"Maid in the Mirror",
+		"Hand",
+		"Void Storm"
+	],
+
+"falling": [
```
Comment (Helg2) [inline strings/phobia.json:207]: non alphabetical too

File: code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm [hunk 4]
```diff
@@ -374,7 +376,9 @@
 	sac_target.remove_status_effect(/datum/status_effect/unholy_determination)
 	sac_target.reagents?.del_reagent(/datum/reagent/inverse/helgrasp/heretic)
 	sac_target.clear_mood_event("shadow_realm")
-	sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/supernatural, TRAUMA_RESILIENCE_MAGIC)
+
```
Comment (Helg2) [inline code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:379]: Linters are probably failing because of a space here

**Review by Helg2 — CHANGES_REQUESTED** [review]

File: code/_globalvars/phobias.dm (hunk 1 above)
Comment (DATA-xPUNGED) [reply code/_globalvars/phobias.dm:27]: I knew i missed one

Comment (Helg2) [inline code/_globalvars/phobias.dm:27]:
```suggestion
```

File: code/_globalvars/phobias.dm (hunk 2 above)
Comment (Helg2) [inline code/_globalvars/phobias.dm:510]:
```suggestion
```
redundant and non-alphabetical

**Comment by nukularpower** [comment]
why not have this linked to the hand dodging minigame?   like dodging some number of hands like 80-100% makes it so you dont get the trauma or something like that

**Comment by DATA-xPUNGED** [comment]
> why not have this linked to the hand dodging minigame? like dodging some number of hands like 80-100% makes it so you dont get the trauma or something like that

The trauma is an replacement for the old interaction where it just fucking gibbed you on sacrifice. So instead it effectively round removes you in the heretic's eyes. With that in mind i don't think that's a good idea

**Comment by DATA-xPUNGED** [comment]
I've got no idea why those unit tests are failing tbh, i didn't touch head.dm

**Comment by DATA-xPUNGED** [comment]
Everyone who i have pestered to try and find out why the unit test isn't working has NO IDEA why it isn't, i give up, I'm manually rebasing this again

**Pull request closed without merging**